File: _2_data_structure/_1_arrays_and_strings/_6_sliding_window/fixed_size_with_simple_calculation/_0_maximum_sum_subarray_of_size_k.cpp
// https://practice.geeksforgeeks.org/problems/max-sum-subarray-of-size-k5313/1

class Solution{   
public:
    long maximumSumSubarray(int K, vector<int> &Arr , int N){
        int i=0,j=0;
        
        long sum = 0,maxi=LONG_MIN;
        
        while(j<N) {
            sum+=Arr[j];
            if(j-i+1==K) {
                maxi=max(maxi, sum);
                sum-=Arr[i];
                i++;
            }
            j++;
        }
        
        return maxi;
    }
};

File: _2_data_structure_new/_1_arrays_and_strings/_6_sliding_window/_1_fixed_size_with_simple_calculation/_0_maximum_sum_subarray_of_size_k/_0_maximum_sum_subarray_of_size_k.java
// https://practice.geeksforgeeks.org/problems/max-sum-subarray-of-size-k5313/1

import java.util.*;

class Solution {
    public static long maximumSumSubarray(int K, ArrayList<Integer> Arr, int N) {
        int i = 0, j = 0;

        long sum = 0, maxi = Long.MIN_VALUE;

        while (j < N) {
            sum += Arr.get(j);
            if (j - i + 1 == K) {
                maxi = Math.max(maxi, sum);
                sum -= Arr.get(i);
                i++;
            }
            j++;
        }

        return maxi;
    }
}
